packages/prisma/src/prisma/schema.prisma

```diff
@@ -188,6 +189,38 @@ model Course {
   updatedAt DateTime @updatedAt
 }
 
+enum LeaderboardType {
+  SESSION_BLOCK
+  SESSION
+  COURSE
+}
+
+model LeaderboardEntry {
+  id Int @id @default(autoincrement())
+
+  type LeaderboardType
+
+  avatar   String?
+  username String
+  score    Float
+
+  participant   Participant @relation(fields: [participantId], references: [id])
+  participantId String      @db.Uuid
+
+  sessionBlock   SessionBlock? @relation(fields: [sessionBlockId], references: [id])
+  sessionBlockId Int?
+
+  session   Session? @relation(fields: [sessionId], references: [id])
+  sessionId String?  @db.Uuid
+
+  course   Course? @relation(fields: [courseId], references: [id])
+  courseId String? @db.Uuid
+
+  @@unique([type, participantId, sessionBlockId])
```

Comment:
three unique indices so that each of the use cases works: leaderboard of a session block, of a session, and of a course